### PR TITLE
PR process and commit message guidelines

### PR DESCRIPTION
**Pull request by g-gaston**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/developer/pull-requests.md
Comment (chrisdoherty4) [inline docs/developer/pull-requests.md:11]: Imposing a 1 commit per PR requirement isn't good. I, for example, manually structure commits and put a lot of thought into them. That effort is wasted and even problematic when I want to go back and perhaps revert a particular change I made.

File: docs/developer/pull-requests.md
Comment (chrisdoherty4) [inline docs/developer/pull-requests.md:10]: I'm OK with advocating not squashing updates until the end but is it really necessary to mandate no re-writing. I feel like that judgement should be left to developers.

To provide an example: if I re-write a function based on feedback why shouldn't I squash that, its a new version implemented differently.

This comes back to the auto removal of LGTM too. If we're auto removing LGTM we're forcing developers to do squashing via the UI else the LGTM is removed - that's not good.

File: docs/developer/pull-requests.md
Comment (g-gaston) [reply docs/developer/pull-requests.md:11]: These changes are not introducing that imposition though, we have prow a GitHub already configured that way. So unless I'm mistaken, even if you don't currently squash your commits manually, they will be automatically squashed before merging (I think I mention this a bit bellow). My main goal was to get the best commit history possible under the current constrains.

That said, if you want to reopen that debate (which includes merge commits, etc.), that's cool with me 🙂

Comment (chrisdoherty4) [reply docs/developer/pull-requests.md:11]: Resolved offline.

File: docs/developer/pull-requests.md
Comment (chrisdoherty4) [reply docs/developer/pull-requests.md:10]: Resolved offline.

**Review by chrisdoherty4 — COMMENTED** [review]
/lgtm

File: docs/developer/pull-requests.md
Comment (TerryHowe) [inline docs/developer/pull-requests.md:11]:
```suggestion
1. Once you haver al least one `lgtm` and are ready to merge, squash your commits into one.
```
```suggestion
1. Once you have al least one `lgtm` and are ready to merge, squash your commits into one.
```
squash on merge no?

Comment (TerryHowe) [inline docs/developer/pull-requests.md:11]:
```suggestion
1. Once you have al least one `lgtm` and are ready to merge, squash your commits into one.
```

Comment (TerryHowe) [reply docs/developer/pull-requests.md:11]: auto squash on merge, no?

File: docs/developer/pull-requests.md
Comment (TerryHowe) [inline docs/developer/pull-requests.md:81]:
```suggestion
A good rule is from Chris Beams on writing a subject is it should complete this sentence:
```
Disturbing connotations of that expression

File: docs/developer/pull-requests.md
Comment (TerryHowe) [reply docs/developer/pull-requests.md:11]: I think we can avoid the squash completely and just use auto squash if we have a good rule on the commit the makes the PR subject because I think the PR subject gets used for the squash, no? I'd have to try it out to be sure.

Comment (TerryHowe) [reply docs/developer/pull-requests.md:11]: Any other options to configure how auto squash works. Nobody has time to squash commits as the final step, I just don't think that is reasonable and while that commit history is not great I just don't think it is worth the extra manual effort to squash.

Comment (chrisdoherty4) [reply docs/developer/pull-requests.md:11]: I, strongly, disagree. I squash manually all the time. `git rebase -i HEAD~<commits to squash>` followed by some vim magic `%s/pick/f/gc` using `gc` options so I can confirm things. 

It literally takes <30s.

Comment (chrisdoherty4) [reply docs/developer/pull-requests.md:11]: That said, this workflow doesn't work with the existing setup because the minute you squash and force push `lgtm` is removed and you're left to bug people for lgtm again. Thats a bad flow.

Comment (g-gaston) [reply docs/developer/pull-requests.md:11]: If losing the `lgtm` label is the only concern, we could use [StoreTreeHash](https://pkg.go.dev/k8s.io/test-infra/prow/plugins#Lgtm) in the prow config. That should prevent the `lgtm` label from being removed if we are only squashing

Comment (chrisdoherty4) [reply docs/developer/pull-requests.md:11]: Lets do it and see where it goes.

**Review by chrisdoherty4 — APPROVED** [review]

**Pull request merged**
